docs/sphinx/source/contributing/style_guide.rst

```diff
@@ -70,6 +70,82 @@ the ``docs/readthedocs.org:pvlib-python`` link within the checks
 status box at the bottom of the pull request.
 
 
+.. _references:
+
+References
+----------
+All pertinent information within a docstring should include a proper reference.
+In pvlib-python, we are transitioning to a standardised referencing system. We
+encourage using the `IEEE style <https://journals.ieeeauthorcenter.ieee.org/wp-content/uploads/sites/7/IEEE_Reference_Guide.pdf>`_
+with numeric in-text citations, but overall the most important feature of all
+references is that they include sufficient information to make locating the
+original source as easy as possible. As a bare minimum, we advise including:
+
+* Author list (can be abbreviated with et al.)
+* Publication title
+* Publication source (journal title, laboratory name, etc.)
+* Year of publication
+* DOI (if available)
+
+The recommended citation style for several media types is as follows:
+
+**Journal article**:
+
+    Author initials. Author Surname, "Title of article," abbreviated journal
+    title, vol. number, issue number, page numbers, Abbreviated Month Year.
+
+**Book**:
+
+    Author initials. Author Surname, “Chapter title” in Book Title, xth ed.
+    City of Publisher, (only U.S. State), Country: Abbrev. of Publisher, year,
+    ch. x, sec. x, pp. xxx–xxx.
+
+**Technical report**:
+
+    Author initials. Author Surname, “Report title” Abbrev. Name of Co.,
+    City of Co., Abbrev. State, Country, Rep. xxx, year
```

Comment:
"Co." can be confused with County. I think we should spell out "Company" or (better) use "publisher" again here. The NREL example does not list the city (Golden, CO).